src/Infrastructure/Services/BindleService.cs

```diff
@@ -83,10 +83,13 @@ public async Task<IEnumerable<string>> GetBindleRevisionNumbers(string bindleId)
 
     public async Task<IEnumerable<string>> QueryAvailableStorages(string query, ulong? offset, int? limit)
     {
-        var matches = await _client.QueryInvoices(query, offset, limit);
+        var matches = await _client.QueryInvoices(query, offset);
```

Comment:
RE: https://github.com/deislabs/hippo/pull/817#issuecomment-1138888779

Totally get what the endpoint tries to do. My question had more to do with why we're removing the `limit` from `QueryInvoices` and how that impacts this endpoint's behaviour.

Re-iterating my earlier question... if you called this endpoint with `limit = 75`, and bindle had 100 invoices that matched that query, wouldn't `_client.QueryInvoices(query, offset)` return 50 invoices because that's [bindle's default search limit](https://github.com/deislabs/bindle/blob/48c672c903216b91a2c0b7f48031788182ec0857/src/search/mod.rs#L28)?

I'm also wondering why we're removing the limit from the query. Wouldn't we want the server to respect the limit query too? e.g. if I requested `limit = 5`, and 1,000,000 invoices matched on the server?

I'm trying to understand why the old code does not work as intended, because it looks fine to me... Can you please provide a test case that reproduces the issue you're seeing?

Reply:
I encountered this issue yesterday when trying to create a new application. It can be reproduced in the following scenario:

1. Push 5 different revisions to the bindle server in the storage "alpha-application"
2. Push a 6th revision using "beta-application" as storage-id
3. Navigate to the 'New Application' page in the Hippo UI and search for ''application"
 
You should notice that the only storage returned would be "alpha-application", even if you would have expected both. 

As a user, if I call the endpoint `QueryAvailableStorages(..., limit = 5)`, I would expect to get less than 5 results only if there actually are less than 5 storages available. In the above case, `_client.QueryInvoices(..., limit = 5)` will return 5 invoices from the "alpha-application" storage so `QueryAvailableStorages(..., limit = 5)` will return only one storage because of the `.Distinct()` clause applied on the storage name, even though in reality there is one more storage ("beta-application") that matches the queryString.

The issue is that the limit is applied to `_client.QueryInvoices(...)` before the `.Distinct()` clause, which results in an undeterministic amount of storages returned.

Indeed, if we do not pass the limit to the bindle client then it will return at most 50 invoices (or less, given the `.Distinct()` clause we are applying). Since we are using the endpoint for pagination, I do not think we will have pages with more than 50 elements, but if you think there might be different use cases for it, we could pass a higher limit to the bindle query.

Please let me know what are your thoughts about the above. Thanks 😄

Reply:
Okay. So if I understand correctly, you're looking for a way to find all storage IDs with unique names. That makes a little more sense now.

Perhaps this is a misunderstanding of the name `QueryAvailableStorages`, as I thought it was intended to search for all invoices regardless if multiple IDs matched.

Can you please add a doc comment above the function describing this function's intended behavior?

Reply:
Yes, that is correct. I have just pushed a commit with the doc comment, please let me know if there are any improvements I can make.

Thank you